reaktive/src/commonTest/kotlin/com/badoo/reaktive/completable/UsingTest.kt

```diff
@@ -0,0 +1,256 @@
+package com.badoo.reaktive.completable
+
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.test.completable.DefaultCompletableObserver
+import com.badoo.reaktive.test.completable.TestCompletable
+import com.badoo.reaktive.test.completable.test
+import com.badoo.reaktive.utils.atomic.AtomicBoolean
+import com.badoo.reaktive.utils.atomic.AtomicInt
+import com.badoo.reaktive.utils.atomic.AtomicList
+import com.badoo.reaktive.utils.atomic.getValue
+import com.badoo.reaktive.utils.atomic.plusAssign
+import com.badoo.reaktive.utils.atomic.setValue
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class UsingTest :
```

Comment:
You can create `class DisposablesFactoryStub: () -> Disposable` that will store all created objects that you can verify later. By doing this we can verify that we invoke `resourceSupplier` only once (or amount of times we needed). Also we can verify that we do not create resource when it is not required.

```kotlin
    @Test
    fun acquires_new_resource_each_time_WHEN_eager_is_true_and_subscribed_multiple_times() {
        val factory = DisposablesFactoryStub()
        val acquiredResources = AtomicList<Disposable>(emptyList())

        val downstream =
            completableUsing(resourceSupplier = factory, eager = true) {
                acquiredResources += it
                TestCompletable()
            }

        repeat(3) { downstream.test() }

        assertContentEquals(factory.created, acquiredResources.value)
    }
```

Reply:
Fixed.